libraries/standard/http/src/http_client.c

```diff
@@ -102,40 +165,204 @@ static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
     HTTPStatus_t returnStatus = HTTP_SUCCESS;
     int32_t transportStatus = 0;
 
-    /* Send the HTTP body over the network. */
-    if( pRequestBodyBuf != NULL )
+    assert( pTransport != NULL );
+    assert( pTransport->send != NULL );
+    assert( pRequestBodyBuf != NULL );
+
+    transportStatus = pTransport->send( pTransport->pContext,
+                                        pRequestBodyBuf,
+                                        reqBodyBufLen );
+
+    if( transportStatus < 0 )
+    {
+        IotLogErrorWithArgs( "Failed to send HTTP body: Transport send() "
+                             " returned error: Transport Status = %d",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else if( transportStatus != reqBodyBufLen )
+    {
+        IotLogErrorWithArgs( "Failed to send HTTP body: Transport send() "
+                             "did not send the required bytes: Required bytes = %d"
+                             ", Sent bytes=%d.",
+                             reqBodyBufLen,
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else
     {
-        transportStatus = pTransport->send( pTransport->pContext,
-                                            pRequestBodyBuf,
-                                            reqBodyBufLen );
+        IotLogDebugWithArgs( "Sent HTTP body over the transport: Bytes sent = %d.",
```

Comment:
In the data section, strings can be compact as "BytesSent=%d" to avoid using extra characters

Reply:
sure